core/vm/contracts.go

```diff
@@ -1048,3 +1064,72 @@ func (c *bls12381MapG2) Run(input []byte) ([]byte, error) {
 	// Encode the G2 point to 256 bytes
 	return g.EncodePoint(r), nil
 }
+
+type remoteStaticCall struct{}
+
+func (c *remoteStaticCall) RequiredGas(input []byte) uint64 {
+	return params.RemoteStaticCallGas
+}
+
+func (c *remoteStaticCall) Run(info PrecompileContext, input []byte) ([]byte, error) {
+	rpcUrl := info.GetL1ArchiveRpc()
+	if rpcUrl == nil {
+		return nil, errors.New("L1 archive RPC is not configured")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	rpcClient, err := rpc.DialContext(ctx, *rpcUrl)
+	ethClient := ethclient.NewClient(rpcClient)
+	cancel()
+	if err != nil {
+		return nil, err
+	}
+
+	L1BlockNumberRaw := info.GetState(types.L1BlockAddr, types.L1BlockNumberSlot)
+	L1BlockNumber := L1BlockNumberRaw.Big()
+
+	// ABI definition to decode `input` data
+	const ethcallAbi = `[{
+		"type" : "function",
+		"name" : "eth_call",
```

Comment:
Ideally the interface to the precompile is `abi.encode(address, bytes)` so we don't need to worry about the 4byte selector being prefixed. We could write a simple solidity library that wraps the abi encode and returns the bytes or even a library that makes the call after abi encoding the args so that devs have a nice interface.

If we wanted to introduce additional functionality like other sorts of calls like fetching L1 balance we could introduce it in a new contract instead of having a precompile with a dispatch table in it based on the 4byte selector

Reply:
Ok I changed it to just be `abi.encode(address, bytes)`, although fwiw I kind of like the idea of having dispatch based on 4 byte selector all to the same address